contrib/exporters/vpclogs/mod/transform.go

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package mod
+
+import (
+	"time"
+
+	cache "github.com/pmylund/go-cache"
+
+	"github.com/spf13/viper"
+
+	"github.com/skydive-project/skydive/contrib/exporters/core"
+	"github.com/skydive-project/skydive/flow"
+	"github.com/skydive-project/skydive/logging"
+)
+
+// NewTransform creates a new flow transformer based on a name string
+func NewTransform(cfg *viper.Viper) (interface{}, error) {
+	excludeStartedFlows := cfg.GetBool(core.CfgRoot + "transform.secadvisor.exclude_started_flows")
+
+	return &vpclogsFlowTransformer{
+		flowUpdateCountCache: cache.New(10*time.Minute, 10*time.Minute),
+		excludeStartedFlows:  excludeStartedFlows,
+	}, nil
+}
+
+const (
+	VpcActionReject string = "R"
+	VpcActionAccept string = "A"
+)
+
+// VpclogsFlow represents a vpc log flow
+// this structure will be updated when the fields are finalized
+type VpclogsFlow struct {
+	FinishType          flow.FlowFinishType `json:"-"`
+	Flow                *flow.Flow          `json:"-"`
+	LastUpdateMetric    *flow.FlowMetric    `json:"-"`
+	Metric              *flow.FlowMetric    `json:"-"`
+	Status              string              `json:"-"`
+	Start               int64               `json:"-"`
+	Protocol            string              `json:"-"`
+	TID                 string              `json:"-"`
+	StartTime           int64               `json:"start_time,omitempty"`
+	EndTime             int64               `json:"end_time,omitempty"`
+	ConnectionStartTime int64               `json:"connection_start_time,omitempty"`
+	// TBD - compute direction
```

Comment:
what do you mean by compute direction?

Reply:
The next field in the structure (according to the spec) is `direction`. The code to compute the direction is not yet written, so this comment is a place-holder to remind us that we still need to implement this.

Reply:
you may consider to add the field but leave it empty (and then if empty don't serialize)

Reply:
Yes, this is what is currently done.